check_usage.py

```diff
@@ -0,0 +1,238 @@
+#!/usr/bin/python
+import argparse
+import datetime
+import time
+
+import urllib2
+import urllib
+import json
+
+docstr = '''
+This script shows user/project usage
+'''
+
+timestamp_format_complete = '%Y-%m-%dT%H:%M:%S'
+timestamp_format_minimal = '%Y-%m-%d'
+
+
+def valid_date(s):
+    '''check if date is in valid format(s)'''
+    complete, minimal = None, None
+
+    try:
+        complete = datetime.datetime.strptime(s, timestamp_format_complete)
+    except ValueError:
+        pass
+
+    try:
+        minimal = datetime.datetime.strptime(s, timestamp_format_minimal)
+    except ValueError:
+        pass
+
+    if not complete and not minimal:  # doesn't fit either format
+        raise argparse.ArgumentTypeError('Not a valid date: {}'.format(s))
+    else:
+        return s
+
+def process_date_time(date_time):
+    try:
+        return time.mktime(time.strptime(date_time, timestamp_format_complete))
+    except ValueError:
+        return time.mktime(time.strptime(date_time, timestamp_format_minimal))
+
+
+
+##### parse arguments #####
+
+
+
+parser = argparse.ArgumentParser(description=docstr)
+
+parser.add_argument('-U', '--user',
+                                help='check usage of this user')
+parser.add_argument('-A', '--account',
+                                help='check usage of this account')
+
+parser.add_argument('-s', '--start', type=valid_date,
+                    help='start time YYYY-MM-DD[THH:MM:SS]'
+                    '(DEFAULT: {}-06-01T00:00:00)'.format(datetime.datetime
+                                                          .now().year),
+                    default='{}-06-01T00:00:00'.format(datetime.datetime
+                                                       .now().year))
+parser.add_argument('-e', '--end', type=valid_date,
+                    help='end time YYYY-MM-DD[THH:MM:SS]'
+                    '(DEFAULT: Current TimeStamp)',
+                    default=datetime.datetime.now()
+                    .strftime(timestamp_format_complete))
+parsed = parser.parse_args()
+user = parsed.user
+account = parsed.account
+_start = parsed.start
+_end = parsed.end
+start = process_date_time(_start)
+end = process_date_time(_end)
+
+if not (user or account):
+    parser.error('Atleast one of -U/--user and -A/--account required.')
+
+
+
+##### prepare params #####
+
+
+
+# base_url = 'https://scgup-dev.lbl.gov:8443/mybrc-rest'
+base_url = 'http://localhost:8880/mybrc-rest'
```

Comment:
It might be helpful to move global variables like this URL to the top of the file for easy changing.

Reply:
good idea :)